utility/src/main/java/org/oppia/util/data/URlImageParser.kt

```diff
@@ -0,0 +1,42 @@
+package org.oppia.util.data
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.LevelListDrawable
+import android.text.Html
+import android.widget.TextView
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.engine.DiskCacheStrategy
+import com.bumptech.glide.request.RequestOptions
+import com.bumptech.glide.request.target.SimpleTarget
+import java.net.URL
+
+/** */
+open class URLImageParser(internal var mTv: TextView, internal var mContext: Context) : Html.ImageGetter {
+
+    override fun getDrawable(source: String): Drawable {
+        val drawable = LevelListDrawable()
+        Glide.with(mContext)
+                .asBitmap()
+                .load(URL(source))
+                .apply( RequestOptions().override(100, 100)) //This is important
+                .diskCacheStrategy(DiskCacheStrategy.RESOURCE)
+                .into(object : SimpleTarget<Bitmap>() {
```

Comment:
I think Glide can provide a Drawable directly per https://bumptech.github.io/glide/doc/getting-started.html#non-view-targets.

Regarding the way this is being done, we could also try something like https://stackoverflow.com/a/7290700/3689782 (though with coroutines rather than AsyncTask), but I actually think we should just go with your solution for now. Soon we'll probably be prefetching the image bitmaps, anyway, to properly support offline support per #169, and that will require changing how this works (the Glide bit will need to move into a domain controller and this code will just be binding a Bitmap that's coming in from ``LiveData``).

Reply:
Ok I have done some changes in the UrlImage Parser I have reformatted code.